pkg/reconciler/isbsvc/installer/watcher.go

```diff
@@ -0,0 +1,36 @@
+package installer
+
+import (
+	"fmt"
+
+	appv1 "k8s.io/api/apps/v1"
+)
+
+// getStatefulSetStatus returns a message describing statefulset status, and a bool value indicating if the status is considered done.
+// Borrowed at kubernetes/kubectl/rollout_status.go https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/rollout_status.go#L130
+func getStatefulSetStatus(sts *appv1.StatefulSet) (string, bool) {
+	if sts.Status.ObservedGeneration == 0 || sts.Generation > sts.Status.ObservedGeneration {
+		return "Waiting for statefulset spec update to be observed...\n", false
+	}
+	if sts.Spec.Replicas != nil && sts.Status.ReadyReplicas < *sts.Spec.Replicas {
```

Comment:
I just want to check what this `ReadyReplicas` means. Does it mean "readiness" in the sense of the "readiness probe", where it means something throughout the life of the Pod, or does it just mean something for when the Pod is starting up? 
Mainly, I want to verify that we're returning false, which is used to set "Progressing", when we are in fact progressing through the initial deployment. 

I know the logic is borrowed from the code you're pointing to above (nice find). Is it being used to indicate something like "progressing" there?

Reply:
ReadyReplicas is the number of pods created for this StatefulSet with a Ready Condition. it for throughout the life of the pod.
"progressing" denotes here either the initial deployment or new pod is coming up, in the reference code they are not mentioned about "Progressing" they just returning status as false/true

Reply:
Thanks @chandankumar4. I'm not sure if I totally understand. So, will `ReadyReplicas` decrease in the case where the StatefulSet was last deployed a long time ago and then something cases one of the Pods to get into a bad state?

Reply:
Yes, For above case `ReadyReplicas` will get decreased.
For example: I have deleted one of the sts pod `isbsvc-default-js-0` and `ReadyReplicas` count decreased to 2 from 3
```
  status:
    availableReplicas: 2
    collisionCount: 0
    currentReplicas: 3
    currentRevision: isbsvc-default-js-597b7f74d7
    observedGeneration: 1
    readyReplicas: 2
    replicas: 3
    updateRevision: isbsvc-default-js-597b7f74d7
    updatedReplicas: 3
```
```
default           isbsvc-default-js-0                        1/3     Running     0             36s
default           isbsvc-default-js-1                        3/3     Running     0             7m7s
default           isbsvc-default-js-2                        3/3     Running     0             7m7s
```

Reply:
Thanks for checking and explaining. So, for Numaplane we need to know “has the ISB Service been fully reconciled?” For that, I would need to be able to distinguish between “just got deployed and not up yet” vs “already got deployed and was initially fine but then the pod started becoming unavailable”. It’s really just the first one I need. That’s what I was thinking “Progressing” might mean. But what do you think?

Reply:
Discussed offline with @chandankumar4 just now. I am fine if we address my request above in a subsequent PR so as to not hold up this one.